doc/man/man3/MasterGui.3
.TH "MasterGui" 3 "Fri Jun 12 2015" "Version 0.0.1" "RIDE" \" -*- nroff -*-
.ad l
.nh
.SH NAME
MasterGui \- 
.SH SYNOPSIS
.br
.PP
.PP
\fC#include <MasterGui\&.h>\fP
.PP
Inherits QMainWindow\&.
.SS "Public Member Functions"

.in +1c
.ti -1c
.RI "\fBMasterGui\fP (QMainWindow *parent=0)"
.br
.ti -1c
.RI "void \fBinitToolBars\fP ()"
.br
.ti -1c
.RI "void \fBinitMenus\fP ()"
.br
.ti -1c
.RI "void \fBsetFileMenuPtr\fP (QMenu *\fBfileMenuPtr\fP)"
.br
.ti -1c
.RI "QMenu * \fBgetFileMenuPtr\fP ()"
.br
.ti -1c
.RI "void \fBsetEditMenuPtr\fP (QMenu *\fBeditMenuPtr\fP)"
.br
.ti -1c
.RI "QMenu * \fBgetEditMenuPtr\fP ()"
.br
.ti -1c
.RI "void \fBsetViewMenuPtr\fP (QMenu *\fBviewMenuPtr\fP)"
.br
.ti -1c
.RI "QMenu * \fBgetViewMenuPtr\fP ()"
.br
.ti -1c
.RI "void \fBsetNavigateMenuPtr\fP (QMenu *\fBnavigateMenuPtr\fP)"
.br
.ti -1c
.RI "QMenu * \fBgetNavigateMenuPtr\fP ()"
.br
.ti -1c
.RI "void \fBsetSourceMenuPtr\fP (QMenu *\fBsourceMenuPtr\fP)"
.br
.ti -1c
.RI "QMenu * \fBgetSourceMenuPtr\fP ()"
.br
.ti -1c
.RI "void \fBsetRefactorMenuPtr\fP (QMenu *\fBrefactorMenuPtr\fP)"
.br
.ti -1c
.RI "QMenu * \fBgetRefactorMenuPtr\fP ()"
.br
.ti -1c
.RI "void \fBsetRunMenuPtr\fP (QMenu *\fBrunMenuPtr\fP)"
.br
.ti -1c
.RI "QMenu * \fBgetRunMenuPtr\fP ()"
.br
.ti -1c
.RI "void \fBsetDebugMenuPtr\fP (QMenu *\fBdebugMenuPtr\fP)"
.br
.ti -1c
.RI "QMenu * \fBgetDebugMenuPtr\fP ()"
.br
.ti -1c
.RI "void \fBsetProfileMenuPtr\fP (QMenu *\fBprofileMenuPtr\fP)"
.br
.ti -1c
.RI "QMenu * \fBgetProfileMenuPtr\fP ()"
.br
.ti -1c
.RI "void \fBsetTeamMenuPtr\fP (QMenu *\fBteamMenuPtr\fP)"
.br
.ti -1c
.RI "QMenu * \fBgetTeamMenuPtr\fP ()"
.br
.ti -1c
.RI "void \fBsetToolsMenuPtr\fP (QMenu *\fBtoolsMenuPtr\fP)"
.br
.ti -1c
.RI "QMenu * \fBgetToolsMenuPtr\fP ()"
.br
.ti -1c
.RI "void \fBsetWindowMenuPtr\fP (QMenu *\fBwindowMenuPtr\fP)"
.br
.ti -1c
.RI "QMenu * \fBgetWindowMenuPtr\fP ()"
.br
.ti -1c
.RI "void \fBsetHelpMenuPtr\fP (QMenu *\fBhelpMenuPtr\fP)"
.br
.ti -1c
.RI "QMenu * \fBgetHelpMenuPtr\fP ()"
.br
.ti -1c
.RI "void \fBsetGroupOneToolBarPtr\fP (QToolBar *\fBgroupOneToolBarPtr\fP)"
.br
.ti -1c
.RI "QToolBar * \fBgetGroupOneToolBarPtr\fP ()"
.br
.ti -1c
.RI "void \fBsetGroupTwoToolBarPtr\fP (QToolBar *\fBgroupTwoToolBarPtr\fP)"
.br
.ti -1c
.RI "QToolBar * \fBgetGroupTwoToolBarPtr\fP ()"
.br
.ti -1c
.RI "void \fBsetGroupThreeToolBarPtr\fP (QToolBar *\fBgroupThreeToolBarPtr\fP)"
.br
.ti -1c
.RI "QToolBar * \fBgetGroupThreeToolBarPtr\fP ()"
.br
.ti -1c
.RI "void \fBsetSouthToolBarPtr\fP (QToolBar *\fBsouthToolBarPtr\fP)"
.br
.ti -1c
.RI "QToolBar * \fBgetSouthToolBarPtr\fP ()"
.br
.ti -1c
.RI "void \fBsetEastToolBarPtr\fP (QToolBar *\fBeastToolBarPtr\fP)"
.br
.ti -1c
.RI "QToolBar * \fBgetEastToolBarPtr\fP ()"
.br
.ti -1c
.RI "void \fBsetWestToolBarPtr\fP (QToolBar *\fBwestToolBarPtr\fP)"
.br
.ti -1c
.RI "QToolBar * \fBgetWestToolBarPtr\fP ()"
.br
.ti -1c
.RI "void \fBsetTabWidget\fP (QTabWidget *\fBtabWidget\fP)"
.br
.ti -1c
.RI "QTabWidget * \fBgetTabWidget\fP ()"
.br
.ti -1c
.RI "void \fBsetEditor\fP (QTextEdit *\fBeditor\fP)"
.br
.ti -1c
.RI "QTextEdit * \fBgetEditor\fP ()"
.br
.ti -1c
.RI "void \fBsetMasterActions\fP (\fBMasterActions\fP *\fBmasterActions\fP)"
.br
.ti -1c
.RI "\fBMasterActions\fP * \fBgetMasterActions\fP ()"
.br
.ti -1c
.RI "void \fBsetMasterToolBars\fP (\fBMasterToolBars\fP *\fBmasterToolBars\fP)"
.br
.ti -1c
.RI "\fBMasterToolBars\fP * \fBgetMasterToolBars\fP ()"
.br
.ti -1c
.RI "void \fBsetMasterMenus\fP (\fBMasterMenus\fP *\fBmasterMenus\fP)"
.br
.ti -1c
.RI "\fBMasterMenus\fP * \fBgetMasterMenus\fP ()"
.br
.ti -1c
.RI "void \fBsetCentralGui\fP (\fBCentralGui\fP *\fBcentralGui\fP)"
.br
.ti -1c
.RI "\fBCentralGui\fP * \fBgetCentralGui\fP ()"
.br
.ti -1c
.RI "\fB~MasterGui\fP ()"
.br
.in -1c
.SS "Private Attributes"

.in +1c
.ti -1c
.RI "QMenu * \fBfileMenuPtr\fP"
.br
.ti -1c
.RI "QMenu * \fBeditMenuPtr\fP"
.br
.ti -1c
.RI "QMenu * \fBviewMenuPtr\fP"
.br
.ti -1c
.RI "QMenu * \fBnavigateMenuPtr\fP"
.br
.ti -1c
.RI "QMenu * \fBsourceMenuPtr\fP"
.br
.ti -1c
.RI "QMenu * \fBrefactorMenuPtr\fP"
.br
.ti -1c
.RI "QMenu * \fBrunMenuPtr\fP"
.br
.ti -1c
.RI "QMenu * \fBdebugMenuPtr\fP"
.br
.ti -1c
.RI "QMenu * \fBprofileMenuPtr\fP"
.br
.ti -1c
.RI "QMenu * \fBteamMenuPtr\fP"
.br
.ti -1c
.RI "QMenu * \fBtoolsMenuPtr\fP"
.br
.ti -1c
.RI "QMenu * \fBwindowMenuPtr\fP"
.br
.ti -1c
.RI "QMenu * \fBhelpMenuPtr\fP"
.br
.ti -1c
.RI "QToolBar * \fBgroupOneToolBarPtr\fP"
.br
.ti -1c
.RI "QToolBar * \fBgroupTwoToolBarPtr\fP"
.br
.ti -1c
.RI "QToolBar * \fBgroupThreeToolBarPtr\fP"
.br
.ti -1c
.RI "QToolBar * \fBsouthToolBarPtr\fP"
.br
.ti -1c
.RI "QToolBar * \fBeastToolBarPtr\fP"
.br
.ti -1c
.RI "QToolBar * \fBwestToolBarPtr\fP"
.br
.ti -1c
.RI "QTabWidget * \fBtabWidget\fP"
.br
.ti -1c
.RI "QTextEdit * \fBeditor\fP"
.br
.ti -1c
.RI "\fBMasterActions\fP * \fBmasterActions\fP"
.br
.ti -1c
.RI "\fBMasterToolBars\fP * \fBmasterToolBars\fP"
.br
.ti -1c
.RI "\fBMasterMenus\fP * \fBmasterMenus\fP"
.br
.ti -1c
.RI "\fBCentralGui\fP * \fBcentralGui\fP"
.br
.in -1c
.SH "Detailed Description"
.PP 
Definition at line 35 of file MasterGui\&.h\&.
.SH "Constructor & Destructor Documentation"
.PP 
.SS "MasterGui::MasterGui (QMainWindow *parent = \fC0\fP)"

.PP
Definition at line 3 of file MasterGui\&.cpp\&.
.SS "MasterGui::~MasterGui ()"

.PP
Definition at line 403 of file MasterGui\&.cpp\&.
.SH "Member Function Documentation"
.PP 
.SS "\fBCentralGui\fP * MasterGui::getCentralGui ()"

.PP
Definition at line 397 of file MasterGui\&.cpp\&.
.SS "QMenu * MasterGui::getDebugMenuPtr ()"

.PP
Definition at line 193 of file MasterGui\&.cpp\&.
.SS "QToolBar * MasterGui::getEastToolBarPtr ()"

.PP
Definition at line 313 of file MasterGui\&.cpp\&.
.SS "QMenu * MasterGui::getEditMenuPtr ()"

.PP
Definition at line 121 of file MasterGui\&.cpp\&.
.SS "QTextEdit * MasterGui::getEditor ()"

.PP
Definition at line 349 of file MasterGui\&.cpp\&.
.SS "QMenu * MasterGui::getFileMenuPtr ()"

.PP
Definition at line 109 of file MasterGui\&.cpp\&.
.SS "QToolBar * MasterGui::getGroupOneToolBarPtr ()"

.PP
Definition at line 265 of file MasterGui\&.cpp\&.
.SS "QToolBar * MasterGui::getGroupThreeToolBarPtr ()"

.PP
Definition at line 289 of file MasterGui\&.cpp\&.
.SS "QToolBar * MasterGui::getGroupTwoToolBarPtr ()"

.PP
Definition at line 277 of file MasterGui\&.cpp\&.
.SS "QMenu * MasterGui::getHelpMenuPtr ()"

.PP
Definition at line 253 of file MasterGui\&.cpp\&.
.SS "\fBMasterActions\fP * MasterGui::getMasterActions ()"

.PP
Definition at line 361 of file MasterGui\&.cpp\&.
.SS "\fBMasterMenus\fP * MasterGui::getMasterMenus ()"

.PP
Definition at line 385 of file MasterGui\&.cpp\&.
.SS "\fBMasterToolBars\fP * MasterGui::getMasterToolBars ()"

.PP
Definition at line 373 of file MasterGui\&.cpp\&.
.SS "QMenu * MasterGui::getNavigateMenuPtr ()"

.PP
Definition at line 145 of file MasterGui\&.cpp\&.
.SS "QMenu * MasterGui::getProfileMenuPtr ()"

.PP
Definition at line 205 of file MasterGui\&.cpp\&.
.SS "QMenu * MasterGui::getRefactorMenuPtr ()"

.PP
Definition at line 169 of file MasterGui\&.cpp\&.
.SS "QMenu * MasterGui::getRunMenuPtr ()"

.PP
Definition at line 181 of file MasterGui\&.cpp\&.
.SS "QMenu * MasterGui::getSourceMenuPtr ()"

.PP
Definition at line 157 of file MasterGui\&.cpp\&.
.SS "QToolBar * MasterGui::getSouthToolBarPtr ()"

.PP
Definition at line 301 of file MasterGui\&.cpp\&.
.SS "QTabWidget * MasterGui::getTabWidget ()"

.PP
Definition at line 337 of file MasterGui\&.cpp\&.
.SS "QMenu * MasterGui::getTeamMenuPtr ()"

.PP
Definition at line 217 of file MasterGui\&.cpp\&.
.SS "QMenu * MasterGui::getToolsMenuPtr ()"

.PP
Definition at line 229 of file MasterGui\&.cpp\&.
.SS "QMenu * MasterGui::getViewMenuPtr ()"

.PP
Definition at line 133 of file MasterGui\&.cpp\&.
.SS "QToolBar * MasterGui::getWestToolBarPtr ()"

.PP
Definition at line 325 of file MasterGui\&.cpp\&.
.SS "QMenu * MasterGui::getWindowMenuPtr ()"

.PP
Definition at line 241 of file MasterGui\&.cpp\&.
.SS "void MasterGui::initMenus ()"

.PP
Definition at line 31 of file MasterGui\&.cpp\&.
.SS "void MasterGui::initToolBars ()"

.PP
Definition at line 75 of file MasterGui\&.cpp\&.
.SS "void MasterGui::setCentralGui (\fBCentralGui\fP *centralGui)"

.PP
Definition at line 391 of file MasterGui\&.cpp\&.
.SS "void MasterGui::setDebugMenuPtr (QMenu *debugMenuPtr)"

.PP
Definition at line 187 of file MasterGui\&.cpp\&.
.SS "void MasterGui::setEastToolBarPtr (QToolBar *eastToolBarPtr)"

.PP
Definition at line 307 of file MasterGui\&.cpp\&.
.SS "void MasterGui::setEditMenuPtr (QMenu *editMenuPtr)"

.PP
Definition at line 115 of file MasterGui\&.cpp\&.
.SS "void MasterGui::setEditor (QTextEdit *editor)"

.PP
Definition at line 343 of file MasterGui\&.cpp\&.
.SS "void MasterGui::setFileMenuPtr (QMenu *fileMenuPtr)"

.PP
Definition at line 103 of file MasterGui\&.cpp\&.
.SS "void MasterGui::setGroupOneToolBarPtr (QToolBar *groupOneToolBarPtr)"

.PP
Definition at line 259 of file MasterGui\&.cpp\&.
.SS "void MasterGui::setGroupThreeToolBarPtr (QToolBar *groupThreeToolBarPtr)"

.PP
Definition at line 283 of file MasterGui\&.cpp\&.
.SS "void MasterGui::setGroupTwoToolBarPtr (QToolBar *groupTwoToolBarPtr)"

.PP
Definition at line 271 of file MasterGui\&.cpp\&.
.SS "void MasterGui::setHelpMenuPtr (QMenu *helpMenuPtr)"

.PP
Definition at line 247 of file MasterGui\&.cpp\&.
.SS "void MasterGui::setMasterActions (\fBMasterActions\fP *masterActions)"

.PP
Definition at line 355 of file MasterGui\&.cpp\&.
.SS "void MasterGui::setMasterMenus (\fBMasterMenus\fP *masterMenus)"

.PP
Definition at line 379 of file MasterGui\&.cpp\&.
.SS "void MasterGui::setMasterToolBars (\fBMasterToolBars\fP *masterToolBars)"

.PP
Definition at line 367 of file MasterGui\&.cpp\&.
.SS "void MasterGui::setNavigateMenuPtr (QMenu *navigateMenuPtr)"

.PP
Definition at line 139 of file MasterGui\&.cpp\&.
.SS "void MasterGui::setProfileMenuPtr (QMenu *profileMenuPtr)"

.PP
Definition at line 199 of file MasterGui\&.cpp\&.
.SS "void MasterGui::setRefactorMenuPtr (QMenu *refactorMenuPtr)"

.PP
Definition at line 163 of file MasterGui\&.cpp\&.
.SS "void MasterGui::setRunMenuPtr (QMenu *runMenuPtr)"

.PP
Definition at line 175 of file MasterGui\&.cpp\&.
.SS "void MasterGui::setSourceMenuPtr (QMenu *sourceMenuPtr)"

.PP
Definition at line 151 of file MasterGui\&.cpp\&.
.SS "void MasterGui::setSouthToolBarPtr (QToolBar *southToolBarPtr)"

.PP
Definition at line 295 of file MasterGui\&.cpp\&.
.SS "void MasterGui::setTabWidget (QTabWidget *tabWidget)"

.PP
Definition at line 331 of file MasterGui\&.cpp\&.
.SS "void MasterGui::setTeamMenuPtr (QMenu *teamMenuPtr)"

.PP
Definition at line 211 of file MasterGui\&.cpp\&.
.SS "void MasterGui::setToolsMenuPtr (QMenu *toolsMenuPtr)"

.PP
Definition at line 223 of file MasterGui\&.cpp\&.
.SS "void MasterGui::setViewMenuPtr (QMenu *viewMenuPtr)"

.PP
Definition at line 127 of file MasterGui\&.cpp\&.
.SS "void MasterGui::setWestToolBarPtr (QToolBar *westToolBarPtr)"

.PP
Definition at line 319 of file MasterGui\&.cpp\&.
.SS "void MasterGui::setWindowMenuPtr (QMenu *windowMenuPtr)"

.PP
Definition at line 235 of file MasterGui\&.cpp\&.
.SH "Member Data Documentation"
.PP 
.SS "\fBCentralGui\fP* MasterGui::centralGui\fC [private]\fP"

.PP
Definition at line 66 of file MasterGui\&.h\&.
.SS "QMenu* MasterGui::debugMenuPtr\fC [private]\fP"

.PP
Definition at line 47 of file MasterGui\&.h\&.
.SS "QToolBar* MasterGui::eastToolBarPtr\fC [private]\fP"

.PP
Definition at line 58 of file MasterGui\&.h\&.
.SS "QMenu* MasterGui::editMenuPtr\fC [private]\fP"

.PP
Definition at line 41 of file MasterGui\&.h\&.
.SS "QTextEdit* MasterGui::editor\fC [private]\fP"

.PP
Definition at line 62 of file MasterGui\&.h\&.
.SS "QMenu* MasterGui::fileMenuPtr\fC [private]\fP"

.PP
Definition at line 40 of file MasterGui\&.h\&.
.SS "QToolBar* MasterGui::groupOneToolBarPtr\fC [private]\fP"

.PP
Definition at line 54 of file MasterGui\&.h\&.
.SS "QToolBar* MasterGui::groupThreeToolBarPtr\fC [private]\fP"

.PP
Definition at line 56 of file MasterGui\&.h\&.
.SS "QToolBar* MasterGui::groupTwoToolBarPtr\fC [private]\fP"

.PP
Definition at line 55 of file MasterGui\&.h\&.
.SS "QMenu* MasterGui::helpMenuPtr\fC [private]\fP"

.PP
Definition at line 52 of file MasterGui\&.h\&.
.SS "\fBMasterActions\fP* MasterGui::masterActions\fC [private]\fP"

.PP
Definition at line 63 of file MasterGui\&.h\&.
.SS "\fBMasterMenus\fP* MasterGui::masterMenus\fC [private]\fP"

.PP
Definition at line 65 of file MasterGui\&.h\&.
.SS "\fBMasterToolBars\fP* MasterGui::masterToolBars\fC [private]\fP"

.PP
Definition at line 64 of file MasterGui\&.h\&.
.SS "QMenu* MasterGui::navigateMenuPtr\fC [private]\fP"

.PP
Definition at line 43 of file MasterGui\&.h\&.
.SS "QMenu* MasterGui::profileMenuPtr\fC [private]\fP"

.PP
Definition at line 48 of file MasterGui\&.h\&.
.SS "QMenu* MasterGui::refactorMenuPtr\fC [private]\fP"

.PP
Definition at line 45 of file MasterGui\&.h\&.
.SS "QMenu* MasterGui::runMenuPtr\fC [private]\fP"

.PP
Definition at line 46 of file MasterGui\&.h\&.
.SS "QMenu* MasterGui::sourceMenuPtr\fC [private]\fP"

.PP
Definition at line 44 of file MasterGui\&.h\&.
.SS "QToolBar* MasterGui::southToolBarPtr\fC [private]\fP"

.PP
Definition at line 57 of file MasterGui\&.h\&.
.SS "QTabWidget* MasterGui::tabWidget\fC [private]\fP"

.PP
Definition at line 61 of file MasterGui\&.h\&.
.SS "QMenu* MasterGui::teamMenuPtr\fC [private]\fP"

.PP
Definition at line 49 of file MasterGui\&.h\&.
.SS "QMenu* MasterGui::toolsMenuPtr\fC [private]\fP"

.PP
Definition at line 50 of file MasterGui\&.h\&.
.SS "QMenu* MasterGui::viewMenuPtr\fC [private]\fP"

.PP
Definition at line 42 of file MasterGui\&.h\&.
.SS "QToolBar* MasterGui::westToolBarPtr\fC [private]\fP"

.PP
Definition at line 59 of file MasterGui\&.h\&.
.SS "QMenu* MasterGui::windowMenuPtr\fC [private]\fP"

.PP
Definition at line 51 of file MasterGui\&.h\&.

.SH "Author"
.PP 
Generated automatically by Doxygen for RIDE from the source code\&.


src/MasterGui.cpp
#include "MasterGui.h"

MasterGui::MasterGui(QMainWindow* parent) : QMainWindow(parent)
{
    this->setWindowIcon(QIcon("/home/james/NetBeansProjects/ride/images/ride1.jpg") );
    this->setWindowTitle("Ride");
    
    masterActions = new MasterActions();
    masterToolBars = new MasterToolBars();
    //openingGui = new OpeningGui();
    masterMenus = new MasterMenus();
    centralGui = new CentralGui();
    centralGui->passNorthTabWidgetPtr(masterActions->getNorthTabWidgetPtr() );
    
    masterToolBars->setMasterActionsPtr(masterActions);
    masterMenus->setMasterActionsPtr(masterActions);
    
    initMenus();
    initToolBars();

    this->setCentralWidget(centralGui);
    masterActions->setNorthTabWidgetPtr(centralGui->getNorthGuiPtr()->getTabWidget() );
    masterActions->setSouthTabWidgetPtr(centralGui->getSouthGuiPtr()/*->getTabWidget()*/ );
    masterActions->setWestWidgetPtr(centralGui->getFileTreeGuiPtr() ); // Pass ("tunnel") method
    masterActions->setEastWidgetPtr(centralGui->getNavigatorGuiPtr() );
    
    this->resize(1000, 800);
}


void MasterGui::initMenus()
{
    fileMenuPtr = menuBar()->addMenu(tr("&File") );
    masterMenus->initFileMenuPtr(fileMenuPtr); 
    
    editMenuPtr = menuBar()->addMenu(tr("&Edit") );
    masterMenus->initEditMenuPtr(editMenuPtr);
    
    viewMenuPtr = menuBar()->addMenu(tr("&View") );
    masterMenus->initViewMenuPtr(viewMenuPtr);
    
    navigateMenuPtr = menuBar()->addMenu(tr("&Navigate") );
    masterMenus->initNavigateMenuPtr(navigateMenuPtr);
    
    sourceMenuPtr = menuBar()->addMenu(tr("&Source") );
    masterMenus->initSourceMenuPtr(sourceMenuPtr);
    
    refactorMenuPtr = menuBar()->addMenu(tr("&Refactor") );
    masterMenus->initRefactorMenuPtr(refactorMenuPtr);
    
    runMenuPtr = menuBar()->addMenu(tr("&Run") );
    masterMenus->initRunMenuPtr(runMenuPtr);
    
    debugMenuPtr = menuBar()->addMenu(tr("&Debug") );
    masterMenus->initDebugMenuPtr(debugMenuPtr);
    
    profileMenuPtr = menuBar()->addMenu(tr("&Profile") );
    masterMenus->initProfileMenuPtr(profileMenuPtr);
    
    teamMenuPtr = menuBar()->addMenu(tr("&Team") );
    masterMenus->initTeamMenuPtr(teamMenuPtr);
    
    toolsMenuPtr = menuBar()->addMenu(tr("&Tools") );
    masterMenus->initToolsMenuPtr(toolsMenuPtr);
    
    windowMenuPtr = menuBar()->addMenu(tr("&Window") );
    masterMenus->initWindowMenuPtr(windowMenuPtr);
    
    helpMenuPtr = menuBar()->addMenu(tr("&Help") );
    masterMenus->initHelpMenuPtr(helpMenuPtr);
    
}


void MasterGui::initToolBars()
{
    groupOneToolBarPtr = addToolBar(tr("New File") );
    addToolBar(Qt::TopToolBarArea, groupOneToolBarPtr);
    masterToolBars->initNorthGroupOneToolBar(groupOneToolBarPtr);
    
    groupTwoToolBarPtr = addToolBar(tr("Undo") );
    addToolBar(Qt::TopToolBarArea, groupTwoToolBarPtr);
    masterToolBars->initNorthGroupTwoToolBar(groupTwoToolBarPtr);

    groupThreeToolBarPtr = addToolBar(tr("Debug") );
    addToolBar(Qt::TopToolBarArea, groupThreeToolBarPtr);
    masterToolBars->initNorthGroupThreeToolBar(groupThreeToolBarPtr);

    southToolBarPtr = addToolBar(tr("South") );
    addToolBar(Qt::BottomToolBarArea, southToolBarPtr);
    masterToolBars->initSouthGroupOneToolBar(southToolBarPtr);
    
    eastToolBarPtr = addToolBar(tr("East") );
    addToolBar(Qt::RightToolBarArea, eastToolBarPtr);
    masterToolBars->initEastGroupOneToolBar(eastToolBarPtr);

    westToolBarPtr = addToolBar(tr("West") );
    addToolBar(Qt::LeftToolBarArea, westToolBarPtr);
    masterToolBars->initWestGroupOneToolBar(westToolBarPtr);
}


void MasterGui::setFileMenuPtr(QMenu* fileMenuPtr)
{
    this->fileMenuPtr = fileMenuPtr;
}


QMenu* MasterGui::getFileMenuPtr()
{
    return fileMenuPtr;
}


void MasterGui::setEditMenuPtr(QMenu* editMenuPtr)
{
    this->editMenuPtr = editMenuPtr;
}


QMenu* MasterGui::getEditMenuPtr()
{
    return editMenuPtr;
}


void MasterGui::setViewMenuPtr(QMenu* viewMenuPtr)
{
    this->viewMenuPtr = viewMenuPtr;
}


QMenu* MasterGui::getViewMenuPtr()
{
    return viewMenuPtr;
}


void MasterGui::setNavigateMenuPtr(QMenu* navigateMenuPtr)
{
    this->navigateMenuPtr = navigateMenuPtr;
}


QMenu* MasterGui::getNavigateMenuPtr()
{
    return navigateMenuPtr;
}


void MasterGui::setSourceMenuPtr(QMenu* sourceMenuPtr)
{
    this->sourceMenuPtr = sourceMenuPtr;
}


QMenu* MasterGui::getSourceMenuPtr()
{
    return sourceMenuPtr;
}


void MasterGui::setRefactorMenuPtr(QMenu* refactorMenuPtr)
{
    this->refactorMenuPtr = refactorMenuPtr;
}


QMenu* MasterGui::getRefactorMenuPtr()
{
    return refactorMenuPtr;
}


void MasterGui::setRunMenuPtr(QMenu* runMenuPtr)
{
    this->runMenuPtr = runMenuPtr;
}


QMenu* MasterGui::getRunMenuPtr()
{
    return runMenuPtr;
}


void MasterGui::setDebugMenuPtr(QMenu* debugMenuPtr)
{
    this->debugMenuPtr = debugMenuPtr;
}


QMenu* MasterGui::getDebugMenuPtr()
{
    return debugMenuPtr;
}


void MasterGui::setProfileMenuPtr(QMenu* profileMenuPtr)
{
    this->profileMenuPtr = profileMenuPtr;
}


QMenu* MasterGui::getProfileMenuPtr()
{
    return profileMenuPtr;
}


void MasterGui::setTeamMenuPtr(QMenu* teamMenuPtr)
{
    this->teamMenuPtr = teamMenuPtr;
}


QMenu* MasterGui::getTeamMenuPtr()
{
    return teamMenuPtr;
}


void MasterGui::setToolsMenuPtr(QMenu* toolsMenuPtr)
{
    this->toolsMenuPtr;
}


QMenu* MasterGui::getToolsMenuPtr()
{
    return toolsMenuPtr;
}


void MasterGui::setWindowMenuPtr(QMenu* windowMenuPtr)
{
    this->windowMenuPtr = windowMenuPtr;
}


QMenu* MasterGui::getWindowMenuPtr()
{
    return windowMenuPtr;
}


void MasterGui::setHelpMenuPtr(QMenu* helpMenuPtr)
{
    this->helpMenuPtr = helpMenuPtr;
}


QMenu* MasterGui::getHelpMenuPtr()
{
    return helpMenuPtr;
}


void MasterGui::setGroupOneToolBarPtr(QToolBar* groupOneToolBarPtr)
{
    this->groupOneToolBarPtr = groupOneToolBarPtr;
}


QToolBar* MasterGui::getGroupOneToolBarPtr()
{
    return groupOneToolBarPtr;
}


void MasterGui::setGroupTwoToolBarPtr(QToolBar* groupTwoToolBarPtr)
{
    this->groupTwoToolBarPtr = groupTwoToolBarPtr;
}


QToolBar* MasterGui::getGroupTwoToolBarPtr()
{
    return groupTwoToolBarPtr;
}


void MasterGui::setGroupThreeToolBarPtr(QToolBar* groupThreeToolBarPtr)
{
    this->groupThreeToolBarPtr;
}


QToolBar* MasterGui::getGroupThreeToolBarPtr()
{
    return groupThreeToolBarPtr;
}


void MasterGui::setSouthToolBarPtr(QToolBar* southToolBarPtr)
{
    this->southToolBarPtr = southToolBarPtr;
}


QToolBar* MasterGui::getSouthToolBarPtr()
{
    return southToolBarPtr;
}


void MasterGui::setEastToolBarPtr(QToolBar* eastToolBarPtr)
{
    this->eastToolBarPtr = eastToolBarPtr;
}


QToolBar* MasterGui::getEastToolBarPtr()
{
    return eastToolBarPtr;
}


void MasterGui::setWestToolBarPtr(QToolBar* westToolBarPtr)
{
    this->westToolBarPtr = westToolBarPtr;
}


QToolBar* MasterGui::getWestToolBarPtr()
{
    return westToolBarPtr;
}


void MasterGui::setTabWidget(QTabWidget* tabWidget)
{
    this->tabWidget = tabWidget;
}


QTabWidget* MasterGui::getTabWidget()
{
    return tabWidget;
}


void MasterGui::setEditor(QTextEdit* editor)
{
    this->editor = editor;
}


QTextEdit* MasterGui::getEditor()
{
    return editor;
}


void MasterGui::setMasterActions(MasterActions* masterActions)
{
    this->masterActions = masterActions;
}


MasterActions* MasterGui::getMasterActions()
{
    return masterActions;
}


void MasterGui::setMasterToolBars(MasterToolBars* masterToolBars)
{
    this->masterToolBars = masterToolBars;
}


MasterToolBars* MasterGui::getMasterToolBars()
{
    return masterToolBars;
}


void MasterGui::setMasterMenus(MasterMenus* masterMenus)
{
    this->masterMenus = masterMenus;
}


MasterMenus* MasterGui::getMasterMenus()
{
    return masterMenus;
}


void MasterGui::setCentralGui(CentralGui* centralGui)
{
    this->centralGui = centralGui;
}


CentralGui* MasterGui::getCentralGui()
{
    return centralGui;
}


QString* MasterGui::toString()
{
    QString* tmp = new QString();
    tmp->append("***method stub***");
    
    return tmp;
}


MasterGui::~MasterGui()
{
    ;
}